### ISSUE TITLE
Should we move things into :client:transport?

### ISSUE THREAD

**Issue by nik9000** [elastic/elasticsearch#19555]
Now that we have the :client:transport project there are two things I'd personally love to do:
1. Move the Builders into it. That way it is super obvious that they are part of the client and not the core. That was something that confused me as a new contributor.
2. Improve way you call modules from the Client. Right now you have to do funky stuff like `ReindexAction.INSTANCE.newRequestBuilder(client())`. It'd be nice if this were as easy as any other action. I think the dependencies are set up so we could do something like that.

Are these good things?

**Comment by nik9000** [comment]
Now that I think about it moving the builders would be a reasonably large change too because NodeClient uses them. Though it looks like for the most part we don't actually use them in Elasticsearch, more in the tests.

**Comment by javanna** [comment]
I think those are good things. This relates to #19055 as we want to try and reuse these classes in the high-level java REST client too.

**Comment by javanna** [comment]
We recently discussed this as part of the planned high level REST client effort. Factoring out the java API builders into a common library is an option, though we are not yet sure how long it is going to take, so we should experiment with that a bit. The main point is that we want to get the high level client out there as soon as possible, hence we might just start with making it depend on core, which is not worse than what the transport client already does, and already much better than in the past as it doesn't depend on guice anymore. The bigger part is probably taking out the `toQuery` method of query builders, as that one depends on lucene. Closing for now as we don't need a separate issue for this, but we will take it into account as part of #19055.
